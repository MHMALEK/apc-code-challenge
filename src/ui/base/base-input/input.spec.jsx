import { shallow } from 'enzyme';

import BaseInput from '.';

describe('BaseInput Componenent render', () => {
  const sampleName = 'sample-name';
  const BaseComponent = shallow(<BaseInput />);
  const BaseComponentWithNameProps = shallow(<BaseInput name={sampleName} />);
  const BaseComponentWithValueProps = shallow(
    <BaseInput value={sampleName} name={sampleName} />,
  );

  it('it should match the SnappShot', () => {
    expect(BaseComponent).toMatchSnapshot();
  });
  it('it should renders the input html element', () => {
    expect(
      BaseComponent.find('input').containsMatchingElement(<input />),
    ).toEqual(true);
  });

  it('it should renders the input with correct name props', () => {
    expect(BaseComponentWithNameProps.find('input').props().name).toEqual(
      sampleName,
    );
  });


  it('it should renders the input with correct value props', () => {
    expect(BaseComponentWithValueProps.find('input').props().value).toEqual(
      sampleName,
    );
  });

  it('it should renders the correct value when onChange handler triggered', () => {
    const onChangeMock = jest.fn();
    const event = {
      preventDefault() {},
      target: { value: 'the-value' },
    };
    const component = shallow(<BaseInput onChange={onChangeMock} />);
    component.find('input').simulate('change', event);
    expect(onChangeMock).toBeCalled();
  });
});
