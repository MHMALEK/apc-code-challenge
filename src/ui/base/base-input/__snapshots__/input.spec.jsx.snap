// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInput Componenent render it should match the SnappShot 1`] = `<input />`;
